devbox/src/syncer/Syncer.scala

```diff
@@ -50,9 +51,15 @@ class Syncer(agent: AgentApi,
     agent.start(s => logger.info(fansi.Color.Blue("Initializing Devbox: ") + s))
     agentActor.spawnReaderThread()
 
+    // let's recover the git index forceIncludes from the skipper (instead of re-reading them from disk)
+    val allForceIncludes: Seq[MutablePathSet] = for (skipper <- skipActor.initialState.skippers) yield skipper match {
```

Comment:
It's not thread-safe to grab the mutable `forceInclude`s from the `skipActor` after it has already been started and is potentially receiving messages and processing them. We also should not share mutable data structures between the actors and the outside code that will be running concurrently and in parallel.

I think the correct thing to do here is to initialize the force-include `MutablePathSet` before the actors start, explicitly copy them to immutable `PathSet`s to be sent as part of the `SyncFiles.RemoteScan` message, and then use the `MutablePathSet`s to initialize the `skipActor`. That way we won't have multiple concurrent threads mangling the same mutable data structure at the same time

Reply:
Yeah, I thought about that, but I thought the actor didn't start yet (the first message send is below) and we created it on this thread, so it should be visible.. but yeah, it's cleaner to not do that so I'll work it out differently.